pkg/api/server/v1alpha2/results_test.go

```diff
@@ -147,6 +149,112 @@ func TestGetResult(t *testing.T) {
+	if err != nil {
+		t.Fatalf("failed to create server: %v", err)
+	}
+	ctx := context.Background()
+
+	tt := []struct {
+		name      string
+		parent    string
+		in        *pb.Result
+		fieldmask *field_mask.FieldMask
+		update    *pb.Result
+		expect    *pb.Result
+		errcode   codes.Code
+	}{
+		{
+			name:   "Test no Mask",
+			in:     &pb.Result{Name: "foo/results/bar-001"},
+			update: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{Annotations: map[string]string{"foo": "bar"}},
+		},
+		{
+			name:      "Test Mask with empty field",
+			in:        &pb.Result{Name: "foo/results/bar-002"},
+			fieldmask: &field_mask.FieldMask{Paths: []string{}},
+			// unset field value to default value in fieldmask
+			update: &pb.Result{Name: "foo/results/bar-002", Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{},
+		},
+		{
+			name:      "Test Mask with nil Paths field",
+			in:        &pb.Result{Name: "foo/results/bar-003"},
+			fieldmask: &field_mask.FieldMask{},
+			// do not update
+			update: &pb.Result{Name: "foo/results/bar-003", Annotations: map[string]string{"foo": "bar"}},
+			expect: &pb.Result{},
+		},
+
+		// Errors
+		{
+			name:      "ERR Test update with invalid name",
+			in:        &pb.Result{Name: "foo/results/bar-005"},
+			fieldmask: &field_mask.FieldMask{Paths: []string{"annotations"}},
+			// do not update
+			update:  &pb.Result{Name: "invalid", Annotations: map[string]string{"foo": "bar"}},
+			errcode: codes.NotFound,
+		},
+		{
+			name:      "ERR Test Mask with invalid mask field",
+			in:        &pb.Result{Name: "foo/results/bar-004"},
+			fieldmask: &field_mask.FieldMask{Paths: []string{"annotations", "invalid_field"}},
+			// do not update
+			update:  &pb.Result{Name: "foo/results/bar-004", Annotations: map[string]string{"foo": "bar"}},
+			errcode: codes.NotFound,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			parent, _, err := result.ParseName(tc.in.GetName())
+			if err != nil {
+				t.Fatalf("could not parse result name: %v", err)
+			}
+			r, err := srv.CreateResult(ctx, &pb.CreateResultRequest{Result: tc.in, Parent: parent})
+			if err != nil {
+				t.Fatalf("could not create taskrun: %v", err)
+			}
+
+			// If we're doing a full update, pass through immutable fields to
+			// the update result. Since these are created dynamicly,
+			// we can't prepopulate these.
+			if tc.fieldmask == nil {
+				tc.update.Id = r.GetId()
+				tc.update.Name = r.GetName()
+				tc.update.CreatedTime = r.GetCreatedTime()
+			}
+			// Update the created taskrun
+			r, err = srv.UpdateResult(ctx, &pb.UpdateResultRequest{Result: tc.update, Name: tc.update.GetName(), UpdateMask: tc.fieldmask})
+			if err != nil {
+				if status.Code(err) != tc.errcode {
+					t.Fatalf("expected error: %v, got %v", tc.errcode, status.Code(err))
+				} else {
+					return
+				}
+				t.Fatalf("could not update taskrun: %v, %v", err, status.Code(err))
```

Comment:
This might be easier to read if we simplify the logic a bit:

```suggestion
				if status.Code(err) == tc.errcode {
					continue
				}
				t.Fatalf("UpdateResult()=(%v, %v); want %v", r, err, tc.errcode)
```

Reply:
Done